tests/prague/eip7692_eof_v1/eip7069_extcall/test_calldata.py

```diff
@@ -0,0 +1,456 @@
+                        Op.EXTSTATICCALL(address_returner, offset, length),
+                    )
+                    + Op.SSTORE(slot_calldata_1, Op.RETURNDATALOAD(0))
+                    + Op.SSTORE(slot_calldata_2, Op.RETURNDATALOAD(32))
+                    + Op.SSTORE(slot_code_worked, value_code_worked)
+                    + Op.STOP
+                ),
+                Section.Data(data=memory),
+            ]
+        ),
+        storage={slot_call_status: value_exceptional_abort_canary},
+        balance=10**9,
+    )
+
+    calldata = memory[offset : offset + length]
+    post = {
+        address_returner: Account(storage={}),
+        address_caller: Account(
+            storage={
+                slot_code_worked: value_code_worked,
+                slot_call_status: EXTCALL_SUCCESS,
+                slot_calldata_1: right_pad_32(calldata[0:32]),
+                slot_calldata_2: right_pad_32(calldata[32:64]),
+            }
+        ),
+    }
+
+    tx = Transaction(to=address_caller, gas_limit=2_000_000, sender=sender)
+
+    state_test(
+        env=env,
+        pre=pre,
+        tx=tx,
+        post=post,
+    )
+
+
+@pytest.mark.parametrize(
+    ["opcode", "keyword_args"],
+    [
+        pytest.param(Op.CALL, {"gas": 500_000}, id="CALL"),
+        pytest.param(Op.CALLCODE, {"gas": 500_000}, id="CALLCODE"),
+        pytest.param(Op.DELEGATECALL, {"gas": 500_000}, id="DELEGATECALL"),
+        pytest.param(Op.STATICCALL, {"gas": 500_000}, id="STATICCALL"),
+        pytest.param(Op.EXTCALL, {}, id="EXTCALL"),
+        pytest.param(Op.EXTDELEGATECALL, {}, id="EXTDELEGATECALL"),
+        pytest.param(Op.EXTSTATICCALL, {}, id="EXTSTATICCALL"),
+    ],
+)
+def test_calldata_remains_after_subcall(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    opcode: Op,
+    keyword_args: Dict[str, Any],
+):
+    """
+    Tests call data remains after a call to another contract.
+
+    Caller pushes data into memory, then calls the target.  Target calls 3rd contract. 3rd contract
+    returns. Target writes calldata to storage.
+    """
+    env = Environment()
+
+    sender = pre.fund_eoa(10**18)
+
+    address_sub_called = pre.deploy_contract(
+        Container(
+            sections=[
+                Section.Code(
+                    code=Op.SSTORE(slot_delegate_code_worked, value_code_worked) + Op.STOP
+                )
+            ]
+        ),
+        storage={slot_delegate_code_worked: value_exceptional_abort_canary},
+    )
+    called_code = (
+        Op.MSTORE(0, value_calldata_2)
+        + Op.SSTORE(slot_call_status, value_exceptional_abort_canary)
+        + Op.SSTORE(slot_calldata_1, value_exceptional_abort_canary)
+        + Op.SSTORE(slot_code_worked, value_exceptional_abort_canary)
+        + Op.SSTORE(
+            slot_call_status,
+            opcode(
+                address=address_sub_called,
+                args_offset=0,
+                args_size=size_calldata,
+                **keyword_args,
```

Comment:
```suggestion
```
`Op.CALL`, `Op.DELEGATECALL`, `Op.STATICCALL` and `Op.CALLCODE` have a default for the `gas` keyword argument of `Op.GAS`, so it can be safely skipped and we can remove this line.

Reply:
Done.  Also, this leaks into the params, which were also updated.